florist/app/jobs/details/page.tsx

```diff
@@ -0,0 +1,242 @@
+    let pillClasses = "status-pill ";
+    let iconName;
+    let statusDescription;
+    switch (String(validStatuses[status])) {
+        case validStatuses.NOT_STARTED:
+            pillClasses += "alert-info";
+            iconName = "radio_button_checked";
+            statusDescription = validStatuses[status];
+            break;
+        case validStatuses.IN_PROGRESS:
+            pillClasses += "alert-warning";
+            iconName = "sync";
+            statusDescription = validStatuses[status];
+            break;
+        case validStatuses.FINISHED_SUCCESSFULLY:
+            pillClasses += "alert-success";
+            iconName = "check_circle";
+            statusDescription = validStatuses[status];
+            break;
+        case validStatuses.FINISHED_WITH_ERROR:
+            pillClasses += "alert-danger";
+            iconName = "error";
+            statusDescription = validStatuses[status];
+            break;
+        default:
+            pillClasses += "alert-secondary";
+            iconName = "";
+            statusDescription = status;
+            break;
+    }
+    return (
+        <div className={pillClasses} id="job-details-status">
+            <i className="material-icons text-sm" id="job-details-status-icon">
+                {iconName}
+            </i>
+            &nbsp;
+            {statusDescription}
+        </div>
+    );
+}
+
+export function JobDetailsServerConfig({ serverConfig }: { serverConfig: string }): ReactElement {
+    const emptyResponse = (
+        <div className="container pt-3 p-0" id="job-details-server-config-empty">
+            Empty.
+        </div>
+    );
+
+    if (!serverConfig) {
+        return emptyResponse;
+    }
+
+    const serverConfigJson = JSON.parse(serverConfig);
+
+    if (typeof serverConfigJson != "object" || Array.isArray(serverConfigJson)) {
+        return (
+            <div className="container pt-3 p-0" id="job-details-server-config-error">
+                Error parsing server configuration.
+            </div>
+        );
+    }
+
+    const serverConfigNames = Object.keys(serverConfigJson);
+
+    if (serverConfigNames.length === 0) {
+        return emptyResponse;
+    }
+
+    return (
+        <div className="container pt-3 p-0">
+            {serverConfigNames.map((serverConfigName, i) => (
+                <div className="row" key={i}>
+                    <div className="col-sm-2" id={`job-details-server-config-name-${i}`}>
+                        <strong className="text-dark">{serverConfigName}:</strong>
+                    </div>
+                    <div className="col-sm" id={`job-details-server-config-value-${i}`}>
+                        {serverConfigJson[serverConfigName]}
+                    </div>
+                </div>
+            ))}
+        </div>
+    );
+}
+
+export function JobDetailsClientsInfo({ clientsInfo }: { clientsInfo: Array<ClientInfo> }): ReactElement {
```

Comment:
I was thinking that since there may be many clients for some experiments, it may be best to have the client info in a table to have clear separation  of rows and columns for easier viewing.

Reply:
I was using divs with bootstrap classes to simulate a table but they don't work the same way. Will switch.